pcdsdevices/elliptec.py

```diff
@@ -0,0 +1,62 @@
+"""
+Classes for ThorLabs Elliptec motors.
+"""
+
+
+from ophyd import Component as Cpt
+from ophyd import Device
+from ophyd.signal import EpicsSignal, EpicsSignalRO
+
+
+class ELL_Base(Device):
+    """
+    Base class for Elliptec stages.
+
+    Sufficient for control of ELL17/20 (28/60mm linear stages) and ELL14/18
+    (rotation) stages.
+
+    Parameters
+    ----------
+    prefix : str
+        The PV base of the stage.
+    """
+    _Target_Position = Cpt(EpicsSignal, ':MOVE', name='Target Position',
+                           kind='normal')
+    _Target_Precision = Cpt(EpicsSignal, ':MOVE.PREC', name='Target Precision',
+                            kind='config')
+    _Target_EGU = Cpt(EpicsSignal, ':MOVE.EGU', name='Target EGU',
+                      kind='config')
+    _Current_Position = Cpt(EpicsSignalRO, ':CURPOS', name='Current Position',
+                            kind='normal')
+    _Current_Precision = Cpt(EpicsSignal, ':CURPOS.PREC',
+                             name='Current Precision', kind='config')
+    _Current_EGU = Cpt(EpicsSignal, ':CURPOS.EGU', name='Current EGU',
+                       kind='config')
+    _Status = Cpt(EpicsSignalRO, ':STATUS', name='Status', kind='normal')
+    _From_Addr = Cpt(EpicsSignal, ':FROM_ADR', name='From Address',
+                     kind='config')
+    _To_Addr = Cpt(EpicsSignal, ':TO_ADR', name='To Address',
+                   kind='config')
+    _Save_Addr = Cpt(EpicsSignal, ':SAVE', name='Save Data', kind='config')
+    _Command = Cpt(EpicsSignal, ':CMD', name='Command', kind='config')
+    _Response = Cpt(EpicsSignalRO, ':RESPONSE', name='Response', kind='config')
```

Comment:
Is it also your intention here to make all of the components hidden from the user?

Reply:
Many of these the casual user will never need. But, I should expose the most useful ones, or provide methods to access them. I'll add this.